meson.build

```diff
@@ -0,0 +1,290 @@
+project(
+  'libmypaint',
+  'c',
+  # API version: see https://github.com/mypaint/libmypaint/wiki/Versioning
+  # See http://semver.org/ for what this means.
+  version: '2.0.0-beta',
+  meson_version: '>=0.60.0',
+  default_options: [
+    'c_std=c99',
+  ],
+)
+
+cc = meson.get_compiler('c')
+
+conf = configuration_data()
+
+pkgconfig = import('pkgconfig')
+
+###############################################################################
+# Project informations
+
+version_full = meson.project_version()
+version_dash_split = version_full.split('-')
+version_stable = version_dash_split[0]
+version_prerelease = version_dash_split.get(1, '') # may be blank
+
+version_array = version_stable.split('.')
+version_major = version_array[0]
+version_minor = version_array[1]
+version_micro = version_array[2]
+
+# The API "platform" or "intercompatibility" version.
+#
+# This one is used for library name prefixes, for introspection
+# namespace versions, for gettext domains, and basically anything that
+# needs to change when backwards or forwards API compatibility changes.
+# Another way of thinking about it: it allows meaningful side by side
+# installations of libmypaint.
+api_platform_version = '@0@.@1@'.format(version_major, version_minor)
+api_name = 'libmypaint-@0@'.format(api_platform_version)
+
+project_url = 'https://github.com/mypaint/libmypaint'
+
+conf.set('PACKAGE_NAME', meson.project_name())
+conf.set('PACKAGE_URL', project_url)
+conf.set('LIBMYPAINT_API_PLATFORM_VERSION', api_platform_version)
+conf.set('LIBMYPAINT_VERSION', version_stable)
+conf.set('LIBMYPAINT_VERSION_FULL', version_full)
+
+gettext_package = api_name
+conf.set_quoted(
+  'GETTEXT_PACKAGE',
+  gettext_package,
+  description: 'The prefix for our gettext translation domains.',
+)
+
+###############################################################################
+# Libtool versioning
+
+# ABI version see: https://autotools.io/libtool/version.html
+# https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+abi_revision = 0 # increment on every release
+abi_current = 0 # inc when add/remove/change interfaces
+abi_age = 0 # inc only if changes backward compat
+# FIXME: Not correct.
```

Comment:
Not sure what is supposed to be incorrect here.

Reply:
It produces the correct `.so` suffix so it is probably fine.

Reply:
It's wrong because that's not how libtool converts those values to soversion. It seems to work only because everything is 0.

I believe [this code](https://github.com/pvanhoof/dir-examples/blob/master/meson-example/meson.build#L52-L54) is the correct method. That repo also shows how you do it if you want to drop libtool-isms as well.